### ISSUE TITLE
Windows Phone 8.1: setOrientation not supported on device

### ISSUE THREAD

**Issue by Defcon0** [gbenvenuti/cordova-plugin-screen-orientation#55]
Hello,

I freaked out when I saw, the plugin now supports Windows Phone 8. I thought, WP 8.1 might also be supported, but no luck :(

Is there any chance that you integrate also some code for the latest Windows Phone bild which is 8.1?

Or *should* it work?

Anyway: Thanks for your great plugin!

Bye Defcon

**Comment by gbenvenuti** [comment]
Hi @Defcon0 ,

I didn't write the WP8 side and don't have a development environment set up.  If you have one and can debug and resolve the issue please submit a pull request.

Unless anyone else can help??

**Comment by Defcon0** [comment]
I'm affraid, I'm not that experienced with cordova/WP 8.1 so that I could do that :( Maybe someone  more experienced could have a look.

**Comment by DeadJoshWalking** [comment]
I think I've worked this out, you can set the autoRotationPreferences straight from Javascript in 8.1 so if you change the setOrientation function in screenorientation.js to:

```javascript
screenOrientation.setOrientation = function (orientation) {
    
    switch (orientation) {
        case 'portrait-primary':
            Windows.Graphics.Display.DisplayInformation.autoRotationPreferences = Windows.Graphics.Display.DisplayOrientations.portrait;
            break;
        case 'landscape-secondary':
            Windows.Graphics.Display.DisplayInformation.autoRotationPreferences = Windows.Graphics.Display.DisplayOrientations.landscapeFlipped;
            break;
        case 'portrait-secondary':
            Windows.Graphics.Display.DisplayInformation.autoRotationPreferences = Windows.Graphics.Display.DisplayOrientations.portraitFlipped;
            break;
        case 'landscape-primary':
            Windows.Graphics.Display.DisplayInformation.autoRotationPreferences = Windows.Graphics.Display.DisplayOrientations.landscape;
            break;
        default:
            Windows.Graphics.Display.DisplayInformation.autoRotationPreferences = Windows.Graphics.Display.DisplayOrientations.portrait;
    }

    // Get the current orientation
    //Windows.UI.ViewManagement.ApplicationView.getForCurrentView().orientation;
};
```

You'll get a really basic version working, the only issue is if you start the app with multiple orientations supported then you'll lose those the first time you set it to something new. You can set the autoRotationPrefences to allow for multiple rotations as follows:

```javascript
Windows.Graphics.Display.DisplayInformation.autoRotationPreferences = [Windows.Graphics.Display.DisplayOrientations.portrait, Windows.Graphics.Display.DisplayOrientations.landscape];
```

But I haven't checked how it works on the other platforms, should it just save the default preferences when it starts?

**Comment by gbenvenuti** [comment]
Hi @DeadJoshWalking,

Each platform has it's own javascript file so you should be able to put this in the .wp8.js file.  You could possiblly check for the JS objects and then if found push to the native cordova exec call.

i.e.

```javascript
if (Windows && Windows.Graphics && Windows.Graphics.Display) {
     // do switch here
} else {
    // exec()
} 
```

If you have a windows environment setup, I'm happy to merge a tested PR.

**Comment by gbenvenuti** [comment]
resolved in #65
